File: inject-java/src/test/groovy/io/micronaut/aop/simple/InvalidInterceptor.java
package io.micronaut.aop.simple;

import io.micronaut.aop.Interceptor;
import io.micronaut.aop.InvocationContext;
import io.micronaut.core.type.Argument;
import io.micronaut.core.type.MutableArgumentValue;

import javax.inject.Singleton;

@Singleton
public class InvalidInterceptor implements Interceptor {

    @Override
    public Object intercept(InvocationContext context) {
        context.getParameters().put("test", MutableArgumentValue.create(Argument.STRING, "value"));
        return context.proceed();
    }
}

File: inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/aop/simple/InvalidInterceptor.kt
package io.micronaut.kotlin.processing.aop.simple

import io.micronaut.aop.Interceptor
import io.micronaut.aop.InvocationContext
import io.micronaut.core.type.Argument
import io.micronaut.core.type.MutableArgumentValue
import jakarta.inject.Singleton

@Singleton
class InvalidInterceptor : Interceptor<Any?, Any?> {

    override fun intercept(context: InvocationContext<Any?, Any?>): Any? {
        context.parameters["test"] = MutableArgumentValue.create(
            Argument.STRING,
            "value"
        )
        return context.proceed()
    }
}
